signer/src/api/info.rs

```diff
@@ -0,0 +1,505 @@
+            }
+        }
+    }
+
+    /// Populates the Bitcoin node tip information from the provided Bitcoin
+    /// client. This uses a combination of `getblockchaininfo` and
+    /// `getnetworkinfo` RPC calls to populate the information.
+    async fn populate_bitcoin_node_info(&mut self, bitcoin_client: &impl BitcoinInteract) {
+        match bitcoin_client.get_blockchain_info().await {
+            Ok(info) => {
+                self.bitcoin.node_chain = Some(info.chain.to_string());
+                self.bitcoin.node_tip = Some(ChainTipInfo {
+                    block_hash: info.best_block_hash.into(),
+                    block_height: info.blocks,
+                });
+            }
+            Err(e) => {
+                tracing::error!("error getting bitcoin node blockchain info: {}", e);
+            }
+        }
+
+        match bitcoin_client.get_network_info().await {
+            Ok(info) => {
+                self.bitcoin.node_version = Some(info.version);
+                self.bitcoin.node_subversion = Some(info.subversion);
+            }
+            Err(e) => {
+                tracing::error!("error getting bitcoin node network info: {}", e);
+            }
+        }
+    }
+
+    /// Populates the Stacks node tip information from the provided Stacks client.
+    /// This uses the `/v2/info` RPC endpoint to populate the information.
+    async fn populate_stacks_node_info(&mut self, stacks_client: &impl StacksInteract) {
+        match stacks_client.get_node_info().await {
+            Ok(node_info) => {
+                self.stacks.node_tip = Some(ChainTipInfo {
+                    block_hash: node_info.stacks_tip.0.into(),
+                    block_height: node_info.stacks_tip_height,
+                });
+                self.stacks.node_bitcoin_block_height = Some(node_info.burn_block_height);
+                self.stacks.node_version = Some(node_info.server_version);
+            }
+            Err(e) => {
+                tracing::error!("error getting stacks node tip: {}", e);
+            }
+        }
+    }
+
+    /// Populates the DKG information from the provided storage.
+    async fn populate_dkg_info(&mut self, storage: &impl DbRead, config: &Settings) {
+        self.dkg.target_rounds = config.signer.dkg_target_rounds.get();
+        self.dkg.min_bitcoin_block_height =
+            config.signer.dkg_min_bitcoin_block_height.map(|h| h.get());
+
+        match storage.get_latest_encrypted_dkg_shares().await {
+            Ok(Some(keys)) => {
+                self.dkg.current_aggregate_key = Some(keys.aggregate_key.to_string());
+                match storage.get_encrypted_dkg_shares_count().await {
+                    Ok(count) => {
+                        self.dkg.rounds = count;
+                    }
+                    Err(e) => {
+                        tracing::error!(
+                            "error reading encrypted DKG shares count from the database: {}",
+                            e
```

Comment:
nit:
```suggestion
                            "error reading encrypted DKG shares count from the database: {e}",
                            
```
Same thing elsewhere.

Reply:
[514757b](https://github.com/stacks-network/sbtc/pull/1251/commits/514757b8db1c5c2c94cbd8a7746e2c3b92a8e7b5)